code/framework/object/src/main/java/io/cattle/platform/object/process/ObjectProcessManager.java

```diff
@@ -25,4 +25,8 @@
 
     ExitReason executeProcess(String processName, Object resource, Map<String, Object> data);
 
+    void scheduleProcessInstanceAndPublishEvent(String processName, Object resource, Map<String, Object> data);
```

Comment:
Thought about it more. Async in name makes sense, as the process gets scheduled in async manner by delegating scheduling to the Deferred Utils instead of scheduling it from the ObjectProcessManager. @sonchang what do you think about the name Craig suggested? I'm so bad when it comes to naming things :-/